Lib/test/support/__init__.py

```diff
@@ -649,20 +654,29 @@ def wrapper(*args, **kw):
     return decorator
 
 
-def requires_hashdigest(digestname):
+def requires_hashdigest(digestname, openssl=None, usedforsecurity=True):
     """Decorator raising SkipTest if a hashing algorithm is not available
 
     The hashing algorithm could be missing or blocked by a strict crypto
     policy.
 
+    If 'openssl' is True, then the decorator checks that OpenSSL provides
+    the algorithm. Otherwise the check falls back to built-in
+    implementations. The usedforsecurity flag is passed to the constructor.
+
     ValueError: [digital envelope routines: EVP_DigestInit_ex] disabled for FIPS
     ValueError: unsupported hash type md4
     """
     def decorator(func):
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
             try:
-                hashlib.new(digestname)
+                if openssl:
+                    if _hashlib is None:
```

Comment:
I would prefer to avoid having to inspect the _hashlib module to check which implementation picked hashlib. But it's not possible right now, so it's ok to rely on the private _hashlib module. My concern was that other Python implementation may want to reuse Python test suite, but may not have a _hashlib module.